storage/src/node/branch.rs

```diff
@@ -326,15 +326,30 @@ pub struct BranchNode {
     /// Element i is the child at index i, or None if there is no child at that index.
     /// Each element is (`child_hash`, `child_address`).
     /// `child_address` is None if we don't know the child's hash.
-    pub children: [Option<Child>; Self::MAX_CHILDREN],
+    pub children: [T; BranchConstants::MAX_CHILDREN],
 }
 
-impl Debug for BranchNode {
+/// Contants used in Branch. Moved out into its own struct as they are
+/// independent of the generic parameter for Branch.
+#[derive(Debug)]
+pub struct BranchConstants;
```

Comment:
I'm generally not a fan of the "constants" type pattern. I prefer to keep the constants close to their owner or where they are used. In this case, `BranchNode` is the appropriate owner because it has the set number of children.

However, in https://github.com/ava-labs/firewood/commit/984123d83b2fc675538e37551edac5a54b2f84f5 (which has caused some gnarly conflicts with your changes here), I created a type alias for the `Children<T>` array. If instead we had a `Children<T>` [newtype](https://doc.rust-lang.org/rust-by-example/generics/new_types.html), we could move the associated constants to that structure.

Reply:
I agree with @demosdemon here. I think the next step is to merge in his changes which add some generics to BranchNode already.